### ISSUE TITLE
Testing refactor

### ISSUE THREAD

**Issue by Daanra** [nunomaduro/larastan#658]
## Description

Currently, the FeaturesTests that we use do not allow you to assert whether Larastan reports an error for a specific piece of code. Instead, we can only test whether Larastan reports no errors. 

It would be if we could add functionality that allows us to assert that Larastan reports certain errors. In addition, it would be nice if we could change Larastan's config during a test such that we can, for example, test functionality that is disabled by default. 

Created a separate issue because of [this comment](https://github.com/nunomaduro/larastan/pull/654#discussion_r484747285_)

**Comment by nunomaduro** [comment]
@Daanra You already have permissions to maintain this package right?

**Comment by Daanra** [comment]
> @Daanra You already have permissions to maintain this package right?

@nunomaduro I do have write permissions for this repository, if that's what you meant.

I could take a stab at this refactor if nobody has any concrete plans for it yet.

**Comment by szepeviktor** [comment]
> concrete plans

![kép](https://user-images.githubusercontent.com/952007/93003145-7a7bd480-f53c-11ea-950e-1c3350bf4c63.png)

**Comment by canvural** [comment]
I'd like to use the `TestCase` classes from PHPStan. The only problem I see is we need to bootstrap Laravel with `testbench` before starting the tests. If we can solve it we can use this approach I think.

**Comment by nunomaduro** [comment]
I am also not happy with the current test suite. Feel free to make a pull request to improve this in the future.
